File: ruby/src/Creational/Abstract_Factory/HonoluluConcreteSoupFactory.rb

#//HonoluluConcreteSoupFactory - Two of Two concrete factories extending the abstract factory

from AbstractSoupFactory import AbstractSoupFactory
from common.HonoluluClamChowder import HonoluluClamChowder
from common.HonoluluFishChowder import HonoluluFishChowder

class HonoluluConcreteSoupFactory(AbstractSoupFactory) :
	def initialize()
		self.factoryLocation = "Honolulu"

	def makeClamChowder(self) : 
		return HonoluluClamChowder()

	def makeFishChowder(self) : 
		return HonoluluFishChowder()

File: csharp/src/Creational/Abstract_Factory/HonoluluConcreteSoupFactory.cs
//HonoluluConcreteSoupFactory.java - Two of Two concrete factories extending the abstract factory

namespace Creational.Abstract_Factory {

using common;

public class HonoluluConcreteSoupFactory : AbstractSoupFactory {
    public HonoluluConcreteSoupFactory() {
        factoryLocation = "Honolulu";
    }
    public new ClamChowder makeClamChowder() {
       return new HonoluluClamChowder();
    }
    public new FishChowder makeFishChowder() {
       return new HonoluluFishChowder();
    }
}

}
